src/main/telemetry/srxl.c

```diff
@@ -194,15 +194,11 @@ uint16_t getMotorAveragePeriod(void)
 
 #if defined(USE_DSHOT_TELEMETRY)
     if (useDshotTelemetry) {
-        uint16_t motors = getMotorCount();
-
-        if (motors > 0) {
-            for (int motor = 0; motor < motors; motor++) {
-                rpm += getDshotTelemetry(motor);
-            }
-            rpm = 100.0f / (motorConfig()->motorPoleCount / 2.0f) * rpm;  // convert erpm freq to RPM.
-            rpm /= motors;           // Average combined rpm
+        rpm = 0;
+        for (int motor = 0; motor < getMotorCount(); motor++) {
+            rpm += getDshotTelemetry(motor);
         }
+        rpm = calcEscRpm(rpm / getMotorCount());
```

Comment:
what if `getMotorCount()` returns zero?

Reply:
I have thought in that. I think there are two possibilities to prevent that situation. Don't allow 0 motors in configurator or checking that possibility before making the calculation (more overload in firmware). Please let me know which option you do prefer.

Reply:
IMO motor count can be zero, when using only servos. But that is very uncommon configuration

Reply:
What configurations use only servos?

Reply:
I mean, what configurations use dshot tememetry but use only servos?

Reply:
Please note that this code path is only executed when dshot erpm telemetry is received, for that to happen getMotorCount is always bigger than 0.

Reply:
@DusKing1 Please note that this code path is only executed when dshot erpm telemetry is received, for that to happen getMotorCount is always bigger than 0.